pkg/controller/apiserver/handlers/handler_test.go

```diff
@@ -68,61 +68,79 @@ var responses = []endpoint.Policies{
 				Name:      "policy2",
 			},
 		},
+		Egress: []antreatypes.NetworkPolicy{},
+		Ingress: []antreatypes.NetworkPolicy{},
 	},
 }
 
-func TestIncompleteOrInvalidArguments(t *testing.T) {
+// TestIncompleteArguments tests how the handler function responds when the user passes in a query command
+// with incomplete arguments (for now, missing pod or namespace)
+func TestIncompleteArguments(t *testing.T) {
 	mockCtrl := gomock.NewController(t)
 	defer mockCtrl.Finish()
 	// sample selector arguments (right now, only supports podname and namespace)
-	podName, namespace := "podName", "namespace"
+	pod, namespace := "pod", "namespace"
 	// outline test cases with expected behavior
 	testCases := map[string]TestCase{
-		"Responds with empty list given no name and no namespace": {
-			handlerArgs:           []string{"", ""},
-			expectedStatus:  http.StatusOK,
-			expectedContent: responses[0],
+		"Responds with error given no name and no namespace": {
+			handlerRequest:           "",
+			expectedStatus:  http.StatusBadRequest,
 			argsMock: []string{"", ""},
-			appliedMock: make([]antreatypes.NetworkPolicy, 0),
-			egressMock: make([]antreatypes.NetworkPolicy, 0),
-			ingressMock: make([]antreatypes.NetworkPolicy, 0),
 		},
-		"Responds with empty list given no name": {
-			handlerArgs:           []string{namespace, ""},
-			expectedStatus:  http.StatusOK,
-			expectedContent: responses[0],
+		"Responds with error given no name": {
+			handlerRequest:           "?namespace=namespace",
+			expectedStatus:  http.StatusBadRequest,
 			argsMock: []string{namespace, ""},
-			appliedMock: make([]antreatypes.NetworkPolicy, 0),
-			egressMock: make([]antreatypes.NetworkPolicy, 0),
-			ingressMock: make([]antreatypes.NetworkPolicy, 0),
 		},
-		"Responds with empty list given no namespace": {
-			handlerArgs:           []string{"", podName},
-			expectedStatus:  http.StatusOK,
-			expectedContent: responses[0],
-			argsMock: []string{"", podName},
-			appliedMock: make([]antreatypes.NetworkPolicy, 0),
-			egressMock: make([]antreatypes.NetworkPolicy, 0),
-			ingressMock: make([]antreatypes.NetworkPolicy, 0),
+		"Responds with error given no namespace": {
+			handlerRequest:           "?pod=pod",
+			expectedStatus:  http.StatusBadRequest,
+			argsMock: []string{"", pod},
+		},
+	}
+
+	evaluateTestCases(testCases, mockCtrl, t)
+
+}
+
+//TODO: need to differentiate the return from invalid and incomplete arguments to give correct instructions to user
```

Comment:
yes `StatusBadRequest` for missing arguments and `StatusNotFound` if the Pod does not exist

Reply:
An empty `Response` will mean that no policies / rules select the endpoint

Reply:
Sounds good